agent/agentssh/agentssh.go

```diff
@@ -763,6 +771,20 @@ func isQuietLogin(rawCommand string) bool {
 	return err == nil
 }
 
+// showServiceBanner will write the service banner if enabled and not blank
+// along with a blank line for spacing.
+func showServiceBanner(session io.Writer, banner *codersdk.ServiceBannerConfig) error {
+	if banner.Enabled && banner.Message != "" {
+		// The banner supports Markdown so we might want to parse it but Markdown is
+		// still fairly readable in its raw form.
+		_, err := io.WriteString(session, banner.Message+"\n\n\r")
```

Comment:
I think we should inspect the newline count at the end of `banner.Message` and at most add two, WDYT? I'm worried we may add too much space because the banner was created with newlines at the end and that could look jarring (note that WebUI allows any number of newlines at the end).

PS. I think we also shouldn't output `\r` here. The SSH session (pty) should (hopefully) already be doing automatic `\n` -> `\r\n` translation if it was requested by the client.

At least on Mac, when I update the service banner in the WebUI, all newlines are simply `\n`, but it'd be possible to post any format via the API. So perhaps we should replace all `\r\n` with `\n` just in case, but we should verify this works as expected on Windows. Otherwise we might want to do dos2unix/unix2dos conversion depending on the platform (sigh).

Reply:
Oh right, the pty isn't active yet at this point. So I misspoke about the `\r\n` translation. Looks like we'll have to replace all `\n` with `\r\n` even on Unix. We do the same with `showMOTD`.

I think you could use a `bufio.NewScanner(strings.Reader(banner.Message))` for convenience, or perhaps a two-way replace (`\n` => `\r\n` and `\r\r\n` => `\r\n`).

Reply:
Good catch!  I completely missed the banner could already have newlines in it.

Reply:
For the extra newline issue I went with a trim before adding the two newlines.